Calculator.c

```diff
@@ -0,0 +1,127 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <ctype.h>
+#include <string.h>
+
+#define MAX 100
+
+typedef struct {
+    int top;
+    int items[MAX];
+} Stack;
+// define top value as -1
+void initStack(Stack *s) {
+    s->top = -1;
+}
+
+// push function
+void push(Stack *s, int value) {
+    if (s->top >= MAX - 1) {
+        printf("stack overflow\n");
+        exit(1);
```

Comment:
Consider returning error codes or exceptions to allow graceful failure.

Reply:
Okay